test/TrancheSpreadTest.t.sol

```diff
@@ -0,0 +1,604 @@
+        ffi[21] = "--bind";
+        ffi[22] = "tranche-reserve-io-ratio-growth='tranche-reserve-io-ratio-linear";
+        ffi[23] = "--bind";
+        ffi[24] = string.concat("tranche-reserve-io-ratio-base=", uint2str(TRANCHE_RESERVE_BASE_IO_RATIO));
+        ffi[25] = "--bind";
+        ffi[26] = string.concat("spread-ratio=", uint2str(SPREAD_RATIO));
+        ffi[27] = "--bind";
+        ffi[28] = string.concat("tranche-space-edge-guard-threshold=", uint2str(TRANCHE_EDGE_THRESHOLD));
+        
+        
+        trancheRefill = bytes.concat(getSubparserPrelude(orderBookSubparser), vm.ffi(ffi));
+    }
+
+    function getTrancheTestSpreadOrder(
+        Vm vm,
+        address orderBookSubparser,
+        uint256 testTrancheSpace,
+        uint256 spreadRatio
+    )
+        internal
+        returns (bytes memory trancheRefill)
+    {
+        string[] memory ffi = new string[](31);
+        ffi[0] = "rain";
+        ffi[1] = "dotrain";
+        ffi[2] = "compose";
+        ffi[3] = "-i";
+        ffi[4] = "src/tranche-spread.rain";
+        ffi[5] = "--entrypoint";
+        ffi[6] = "calculate-io";
+        ffi[7] = "--entrypoint";
+        ffi[8] = "handle-io";
+        ffi[9] = "--bind";
+        ffi[10] = "distribution-token=0xd0e9c8f5Fae381459cf07Ec506C1d2896E8b5df6";
+        ffi[11] = "--bind";
+        ffi[12] = "reserve-token=0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270";
+        ffi[13] = "--bind";
+        ffi[14] = "get-tranche-space='get-test-tranche-space";
+        ffi[15] = "--bind";
+        ffi[16] = "set-tranche-space='set-test-tranche-space";
+        ffi[17] = "--bind";
+        ffi[18] = string.concat("test-tranche-space=", uint2str(testTrancheSpace));
+        ffi[19] = "--bind";
+        ffi[20] = "tranche-reserve-amount-growth='tranche-reserve-amount-growth-constant";
+        ffi[21] = "--bind";
+        ffi[22] = string.concat("tranche-reserve-amount-base=", uint2str(TRANCHE_RESERVE_BASE_AMOUNT));
+        ffi[23] = "--bind";
+        ffi[24] = "tranche-reserve-io-ratio-growth='tranche-reserve-io-ratio-linear";
+        ffi[25] = "--bind";
+        ffi[26] = string.concat("tranche-reserve-io-ratio-base=", uint2str(TRANCHE_RESERVE_BASE_IO_RATIO));
+        ffi[27] = "--bind";
+        ffi[28] = string.concat("spread-ratio=", uint2str(spreadRatio));
+        ffi[29] = "--bind";
+        ffi[30] = string.concat("tranche-space-edge-guard-threshold=", uint2str(TRANCHE_EDGE_THRESHOLD));
+        
+        
+        trancheRefill = bytes.concat(getSubparserPrelude(orderBookSubparser), vm.ffi(ffi));
+    }
+
+    function getSubparserPrelude(address obSubparser) internal pure returns (bytes memory) {
+        bytes memory RAINSTRING_OB_SUBPARSER =
+            bytes(string.concat("using-words-from ", obSubparser.toHexString(), " "));
+        return RAINSTRING_OB_SUBPARSER;
+    }
+    
+    function uint2str(uint256 _i) internal pure returns (string memory _uintAsString) {
```

Comment:
this already exists in open zeppelin, which is a dep of this repo

Reply:
resolved